### ISSUE TITLE
WoR identify cheeze

### ISSUE THREAD

**Issue by igroglaz** [draconisPW/PWMAngband#557]
[10:22] YAROSSS: Old bug with scrl of recall 
When it's un Id and read it asks to confirm recall (if the dungeon lvl is lower then player lvl) so I can press (no) and easily id it

**Comment by igroglaz** [comment]
[23:39] PowerWyrm: WoR is easy_know, there's no situation where reading it don't id it so that's not an issue

**Comment by igroglaz** [comment]
IMHO it's a minor exploit.. as by default behaviour you can't ID WoR in such way. But with the option 
`context->origin->player, confirm_recall`
which were recently introduced in PWMA - now you can ID WoR and cancel it...


I suppose legit way for confirm_recall will be auto-using WoR for the 1st time (if it wasn't IDed before) without confirm_recall prompt.. but if player already know WoR scrolls - then confirm_recall should be triggered. Though I'm not sure how to check if scroll were unknown for player before in `bool effect_handler_RECALL(effect_handler_context_t *context)`.. so for now 
I'll have to comment out confirm_recall from T :(

**Comment by igroglaz** [comment]
I've tried this:
```
        // Ask for confirmation if we try to recall from non-reentrable dungeon
        // (added  p->was_aware to check is item IDed 1st time or not - to fix ID cheez by 'no')
        if ((context->origin->player->current_value == ITEM_REQUEST) &&
            OPT(context->origin->player, confirm_recall) &&
            forbid_reentrance(context->origin->player) && context->origin->player->was_aware)
        {
            get_item(context->origin->player, HOOK_CONFIRM, "");
            return false;
        }
```
but it seems 'was_aware' works in different way

**Comment by draconisPW** [comment]
As commented above, there can't be any exploit as whatever you do the scroll will be identified, whether you get a confirmation or not. Getting a confirmation IDs the scroll one second before it gets IDed anyway? Big deal...

**Comment by igroglaz** [comment]
The problem is: you can ID scroll safely by cancelling it action. In normal situation you won't be able to do it and will get into 'interesting' situation.

Right now:
1) you read unIDed scroll
2) you see that it's WoR scroll by `confirm_recall` option and cancel it action
3) Cheeze:
--- you have IDed scroll
--- you have this scroll in you backpack 
--- you stay on the samm spot

Should be:
1) you read unIDed scroll
2) you see that it's WoR scroll BECAUSE it's spent (dissapeared)... and now you have roguelikish situation which you have to solve - WoR spent and in some time you will go outside of the dungeon, so you might want to read another WoR scroll to cancel it. 
3) Why it's fair:
--- you've spend WoR scroll unwillingly
--- most likely you will have to spend yet another scroll to cancel 1st WoR scroll
--- if you didn't notice it - you will be teleported out (roguelikish experience)

**Comment by draconisPW** [comment]
Doesn't matter, the scroll is IDed automatically, whatever happens next is totally out of scope. Same with enchanting scrolls, ID scrolls, remove curse scrolls and so on. If you decide to cancel enchantment, ID and so on, the scroll is still IDed. Your list above is flawed, as ID happens between 1 and 2, not after 2.

**Comment by igroglaz** [comment]
It may be out of scope for PWMA, but in T it makes unfair advantage. It's like introduce option "confirm_death"... so when character die - you will be able to cancel it XD Anyway, in T I'll just comment it out

**Comment by igroglaz** [comment]
pretty good solution: 
[15:36] YAROSSS: Maybe make option to confirm recall be usable only after lvl 10 or 15?
